### PR TITLE
Feature3 - add lessons and terms

### PR DESCRIPTION
**Pull request by jmkoni**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,50 +5,44 @@ git_source(:github) do |repo_name|
...
 gem 'sass-rails', '~> 5.0'
-# Use Uglifier as compressor for JavaScript assets
-gem 'uglifier', '>= 1.3.0'
-# Use CoffeeScript for .coffee assets and views
-gem 'coffee-rails', '~> 4.2'
-# See https://github.com/rails/execjs#readme for more supported runtimes
-# gem 'therubyracer', platforms: :ruby
-
-# Use jquery as the JavaScript library
-gem 'jquery-rails'
```
Comment (osondoar) [inline Gemfile:25]: We might need jquery in the future. But 👍 for removing it for now.

File: app/models/lesson.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Lesson < ApplicationRecord
+  has_and_belongs_to_many :terms
+  validates_presence_of :body, :lesson_type, :questions, :title
+  validates_inclusion_of :lesson_type, in: ["Activity-based", "Article-based"]
```
Comment (osondoar) [inline app/models/lesson.rb]: I would use snake case for the types : `[:activity_based, :article_base]` (or maybe `['activity-based', 'article_based'] `?)

File: Gemfile [hunk 3]
```diff
@@ -5,50 +5,44 @@ git_source(:github) do |repo_name|
...
-# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
 gem 'turbolinks', '~> 5'
-# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 2.5'
```
Comment (osondoar) [inline Gemfile:29]: removing jbuilder was probably unintentional? It looks like some views use it.

File: db/migrate/20170519135150_create_lessons.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class CreateLessons < ActiveRecord::Migration[5.0]
+  def change
+    create_table :lessons do |t|
+      t.text :byline
+      t.string :lesson_type
```
Comment (osondoar) [inline db/migrate/20170519135150_create_lessons.rb:5]: Totally optional: What if we use the postgres enum type instead for `lesson_type`? Rails supports it with `enum lesson_type: { type0: 0, type1: 1 }` in the model.

File: db/migrate/20170519135150_create_lessons.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      t.text :terms, array: true, default: []
```
Comment (osondoar) [inline db/migrate/20170519135150_create_lessons.rb:6]: pg arrays! 😍

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -42,14 +43,15 @@
   # You can disable this behaviour by removing the line below, and instead
   # explicitly tag your specs with their type, e.g.:
   #
-  #     RSpec.describe UsersController, :type => :controller do
```
Comment (osondoar) [inline spec/rails_helper.rb:45]: The type is `controller`, but as this is a comment it really doesn't matter.

**Comment by osondoar** [comment]
All done with my review. Good job! 🚀

**Review by almightyboz — COMMENTED** [review]
👍 :shipit:

**Pull request merged**